src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/GetInstanceBatchesByTimeStamp.sql

```diff
@@ -0,0 +1,52 @@
+/***************************************************************************************/
+-- STORED PROCEDURE
+--     GetInstanceBatchesByTimeStamp
+--
+-- DESCRIPTION
+--     Divides up the instances into a configurable number of batches filter by timestamp.
+--
+-- PARAMETERS
+--     @batchSize
+--         * The desired number of instances per batch. Actual number may be smaller.
+--     @batchCount
+--         * The desired number of batches. Actual number may be smaller.
+--     @status
+--         * The instance status.
+--     @startTimeStamp
+--         * The start filter timestamp.
+--     @endTimeStamp
+--         * The end filter timestamp.
+--     @maxWatermark
+--         * The optional inclusive maximum watermark.
+--
+-- RETURN VALUE
+--     The batches as defined by their inclusive minimum and maximum values.
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.GetInstanceBatchesByTimeStamp
```

Comment:
Can this sproc be used instead of [`GetInstancesByWatermarkRangeV6`](https://github.com/microsoft/dicom-server/blob/main/src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/GetInstancesByWatermarkRangeV6.sql) where the timestamps are optional like `maxWatermark`?

Reply:
I don't think that is the sproc, we could reuse the dbo.GetInstanceBatches with optional timestamp. I am thinking of fixing the customer and refactoring later for perfection, if you insist reusing I put the effort now.